File: java_programming/lab_2/Trade.java
class Trade {
	private String id;
	private String symbol;
	private int quantity;
	private double price;

	public Trade(String id, String symbol, int quantity, double price) {
		this.id = id;
		this.symbol = symbol;
		this.quantity = quantity;
		this.price = price;
	}

	public Trade(String id, String symbol, int quantity) {
		this(id,symbol,quantity,0);
	}

	public void setPrice(double newPrice) {
		if(newPrice < 0) return;
		this.price = newPrice;
	}

	public double get() {
	}

	@Override
	public String toString() {
		return "Trade("+id+","+symbol+","+quantity+","+price+")";
	}
}

File: scala/lab_3/src/main/scala/trade/Trade.scala
package io.turntabl
package trade

case class Trade(val id: String, val symbol: String, val quantity: Int, initialPrice: Double) {
  private var _price = initialPrice

  def price: Double = _price

  def price_= (newPrice: Double): Unit = {
    if newPrice >= 0 then _price = newPrice
  }

  def value: Double = quantity * price

//  override def toString = s"Trade($id, $symbol, $quantity, $price)"
}

//object Trade {
//  def apply(id: String, symbol: String,
//                    quantity: Int, initialPrice: Double): Trade =
//    new Trade(id, symbol, quantity, initialPrice)
//}